diff --git forkSrcPrefix/Makefile forkDstPrefix/Makefile
index 063e5f4d51518fe706c5ee1d8f3a11980c3bc1f9..93db36d0fa5bcd810cc8275e49fd0adfe2fb9247 100644
--- forkSrcPrefix/Makefile
+++ forkDstPrefix/Makefile
@@ -482,6 +482,19 @@ else ifeq ($(platform), miyoo)
 	HAVE_DYNAREC := 1
 	CPU_ARCH := arm
 
+else ifeq ($(platform), rg353)
+  TARGET := $(TARGET_NAME)_libretro.so
+  CC  = $(CROSS_COMPILE)gcc
+  CXX = $(CROSS_COMPILE)g++
+  AR  = $(CROSS_COMPILE)ar
+  SHARED := -shared -Wl,--version-script=link.T
+  fpic := -fPIC
+  CFLAGS += -fomit-frame-pointer -ffast-math -mtune=cortex-a55 -march=armv8.2-a 
+  LDFLAGS += -Wl,--no-undefined
+  HAVE_DYNAREC := 1
+  MMAP_JIT_CACHE = 1
+  CPU_ARCH := arm64
+
 # Windows
 else
 	TARGET := $(TARGET_NAME)_libretro.dll
