diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index 7b9618e1ccfc63982be50b82fda498d75e307fdc..6fae9d28a2867214e3cae29b2abbe1f9a6bafbe2 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -490,6 +490,25 @@ else ifeq ($(platform), miyoo)
 	DYNAREC = ari64
 	HAVE_NEON = 0
 
+# RG35XX
+else ifeq ($(platform), rg35xx)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   # CFLAGS += -marm -mtune=cortex-a9 -mfpu=vfpv3-d16 -mfloat-abi=hard -march=armv7-a
+   CFLAGS += -marm -mtune=cortex-a9 -mfpu=neon-fp16 -mfloat-abi=hard -march=armv7-a -fomit-frame-pointer
+   CFLAGS += -ffast-math -fdata-sections -ffunction-sections -fsingle-precision-constant -flto -fPIC -DNO_HWCAP 
+   LDFLAGS += -flto -fPIC
+   LDFLAGS += -flto
+   CPU_ARCH := arm
+   OPTIMIZE += -Ofast -DNDEBUG=1
+   HAVE_NEON = 1
+   HAVE_NEON_ASM = 1
+   BUILTIN_GPU = neon
+   ARCH = arm
+   DYNAREC = ari64
+
 # Emscripten
 else ifeq ($(platform), emscripten)
    TARGET  := $(TARGET_NAME)_libretro_$(platform).bc
