diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index 036c41f48b087c99b2276b2d41a71d661fd85145..891eb96ceeb1da868d6d2ed40392b51b65dcbb20 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -461,6 +461,23 @@ else ifeq ($(platform), miyoo)
 	PLATFORM_DEFINES += -fomit-frame-pointer -ffast-math -mcpu=arm926ej-s
 	EXTERNAL_ZLIB = 1
 
+# MIYOO FLIP
+else ifeq ($(platform), my355)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--version-script=src/drivers/libretro/link.T -Wl,-no-undefined
+	LDFLAGS += -fPIC -flto
+	CFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 -march=armv8.2-a 
+	CFLAGS += -fomit-frame-pointer -ffast-math -fPIC -flto
+	OPTIMIZE := -Ofast -DNDEBUG
+	EXTERNAL_ZLIB = 1
+	HAVE_NTSC = 0
+	ifeq (,$(DEBUG))
+		# LDFLAGS += -s
+	endif
+	
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 	CC  = cl.exe
