diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index 0d1dc08f81a11cef6b3b7238367b1ff27fcf1ead..1f87c76c3242023c4095f0c9087c52765d93c33c 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -497,6 +497,30 @@ else ifeq ($(platform), miyoo)
 	HAVE_ARMv6 = 0
 	LOW_MEMORY = 1
 
+# MIYOO FLIP
+else ifeq ($(platform), my355)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--no-undefined
+	DONT_COMPILE_IN_ZLIB = 1
+	CFLAGS += -Ofast \
+	-fdata-sections -ffunction-sections -Wl,--gc-sections \
+	-fno-stack-protector -fno-ident -fomit-frame-pointer \
+	-falign-functions=1 -falign-jumps=1 -falign-loops=1 \
+	-fno-unwind-tables -fno-asynchronous-unwind-tables -fno-unroll-loops \
+	-fmerge-all-constants -fno-math-errno
+	CFLAGS += -fPIC -flto
+	CFLAGS += -fomit-frame-pointer -ffast-math -D__GCW0__
+	CFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 -march=armv8.2-a 
+	LDFLAGS += -flto
+	HAVE_NEON = 1
+	BUILTIN_GPU = neon
+	ifeq (,$(DEBUG))
+		LDFLAGS += -s
+	endif
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 	NO_GCC := 1
