diff --git forkSrcPrefix/Makefile forkDstPrefix/Makefile
index dc4b60dec7f70a153dbc2eb849cd5acfc90f9d4d..4ef410b9999fdd744a8f9906a54122984cbe7fe1 100644
--- forkSrcPrefix/Makefile
+++ forkDstPrefix/Makefile
@@ -498,6 +498,18 @@ else ifeq ($(platform), miyoo)
 	HAVE_DYNAREC := 1
 	CPU_ARCH := arm
 	
+else ifeq ($(platform), rgb30)
+  TARGET := $(TARGET_NAME)_libretro.so
+  CC  = $(CROSS_COMPILE)gcc
+  CXX = $(CROSS_COMPILE)g++
+  AR  = $(CROSS_COMPILE)ar
+  SHARED := -shared -Wl,--version-script=link.T
+  fpic := -fPIC
+  CFLAGS += -fomit-frame-pointer -ffast-math -mtune=cortex-a55 -march=armv8.2-a 
+  LDFLAGS += -Wl,--no-undefined
+  HAVE_DYNAREC := 1
+  MMAP_JIT_CACHE = 1
+  CPU_ARCH := arm64
 
 else ifeq ($(platform), miyoomini)
 	TARGET := $(TARGET_NAME)_plus_libretro.so
