diff --git forkSrcPrefix/platform/libretro/Makefile forkDstPrefix/platform/libretro/Makefile
index 40805b8adf8101c90e68c5418e7ea697905562ac..c57d3c28dd22040f19299402a0d569f32585fcf0 100644
--- forkSrcPrefix/platform/libretro/Makefile
+++ forkDstPrefix/platform/libretro/Makefile
@@ -123,7 +123,16 @@ else ifeq ($(platform), miyoomini)
    CXXFLAGS += -marm -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard -march=armv7ve -D_NEED_FULL_PATH_
    fpic := -fPIC
    SHARED := -shared -Wl,--version-script=link.T
-
+else ifeq ($(platform), trimuismart)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   STRIP = $(CROSS_COMPILE)strip
+   CXXFLAGS += -marm -mtune=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard -mcpu=cortex-a7
+   CXXFLAGS += -fomit-frame-pointer -ffast-math -fno-common -ftree-vectorize -funswitch-loops
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=link.T
 else ifeq ($(platform), gcw0)
    TARGET := $(TARGET_NAME)_libretro_gcw0.so
    CC = /opt/gcw0-toolchain/usr/bin/mipsel-linux-gcc
diff --git forkSrcPrefix/platform/libretro/libretro.cpp forkDstPrefix/platform/libretro/libretro.cpp
index bc160dba6d676c7f9c05768f002ea6bce8fe8437..cfa237d9763e2c7092d83aa9fd346c14e4e79db8 100644
--- forkSrcPrefix/platform/libretro/libretro.cpp
+++ forkDstPrefix/platform/libretro/libretro.cpp
@@ -517,6 +517,7 @@ EXPORT bool retro_load_game(struct retro_game_info const *info)
     else {
         _vm->QueueCartChange(info->path);
     }
+	_vm->UpdateAndDraw();
 
     return true;
 }
