diff --git a/libretro/libretro.c b/libretro/libretro.c
index 1f8b3c2b46a79372860c928a47a840b18cc60710..9e849f174c05e78c7808a63ad151db64ab4c3b87 100644
--- a/libretro/libretro.c
+++ b/libretro/libretro.c
@@ -1007,6 +1007,45 @@ void retro_init (void)
 
 ///////////////////////////////////////////////////////////
 
+static int frames = 0;
+static int defer_frames = 36;
+static void* deferred_data = NULL;
+static size_t deferred_size = 0;
+
+static void free_deferred_data(void) {
+	if (deferred_size) {
+		if (log_cb) log_cb(RETRO_LOG_INFO, "Freed deferred save state.\n");
+		free(deferred_data);
+		deferred_data = NULL;
+		deferred_size = 0;
+	}
+}
+
+static void defer_unserialize(const void *data, size_t size) {
+	free_deferred_data();
+
+	if (data==NULL || size<=0) return;
+
+	if (log_cb) log_cb(RETRO_LOG_INFO, "Deferred save state load.\n");
+
+	deferred_data = malloc(size);
+	memcpy(deferred_data, data, size);
+	deferred_size = size;
+}
+static void update_deferred(void) {
+	frames += 1;
+
+	if (frames<defer_frames || deferred_size==0) return;
+
+	retro_unserialize(deferred_data, deferred_size);
+
+	if (log_cb) log_cb(RETRO_LOG_INFO, "Loaded deferred save state.\n");
+
+	free_deferred_data();
+}
+
+///////////////////////////////////////////////////////////
+
 void retro_deinit(void)
 {
 	DeactivateControllerRumble();
@@ -1029,6 +1068,8 @@ void retro_deinit(void)
 	turbo_counter     = 0;
 
 	update_av_info = false;
+	
+	free_deferred_data();
 }
 
 ///////////////////////////////////////////////////////////
@@ -1045,6 +1086,8 @@ void retro_reset (void)
 
 void retro_run (void)
 {
+	if (frames<defer_frames) update_deferred();
+	
 	size_t audio_samples_per_frame;
 	
 	// Check for core options updates
@@ -1223,6 +1266,11 @@ bool retro_serialize(void *data, size_t size)
 
 bool retro_unserialize(const void *data, size_t size)
 {
+	if (frames<defer_frames) {
+		defer_unserialize(data,size);
+		return true;
+	}
+
 	if (PokeMini_LoadSSStream((uint8_t*)data, size)) {
 		if (log_cb) log_cb(RETRO_LOG_INFO, "Save state loaded successfully.\n");
 	} else {
diff --git a/source/PokeMini.c b/source/PokeMini.c
index 06140261ba4d4c36e9605ecfee471bcfddba5f25..9b6f6912185cc1017c129c1d5953688fe5e0019a 100644
--- a/source/PokeMini.c
+++ b/source/PokeMini.c
@@ -489,7 +489,7 @@ int PokeMini_LoadSSStream(uint8_t *buffer, uint64_t size)
 				memstream_set_buffer(NULL, 0);
 				return 0;
 			}
-		} else if (!strcmp(PMiniStr, "LCD-")) {		// Audio
+		} else if (!strcmp(PMiniStr, "AUD-")) {		// Audio
 			if (!MinxAudio_LoadStateStream(stream, BSize)) {
 				memstream_close(stream);
 				memstream_set_buffer(NULL, 0);
