diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index 7b9618e1ccfc63982be50b82fda498d75e307fdc..f314adcbea17f05f473e1cbafc8711ab8ea0c5f1 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -391,6 +391,25 @@ else ifeq ($(platform), rpi4_64)
         fpic := -fPIC
         CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a72 -ftree-vectorize
 
+# MIYOO FLIP
+else ifeq ($(platform), my355)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	CFLAGS += -mcpu=cortex-a55 -mtune=cortex-a55 -march=armv8.2-a  -fomit-frame-pointer
+	CFLAGS += -ffast-math -fdata-sections -ffunction-sections -fsingle-precision-constant -flto -fPIC
+	LDFLAGS += -flto -fPIC
+	ifeq (,$(DEBUG))
+		# LDFLAGS += -s
+	endif
+	CPU_ARCH := arm64
+	OPTIMIZE := -Ofast -DNDEBUG=1
+	HAVE_NEON = 1
+	BUILTIN_GPU = neon
+	ARCH = arm64
+	DYNAREC = ari64
+
 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>
 # Help at https://modmyclassic.com/comp
