diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index ffaba21e7a88f90786d818ab550b68acd05738bf..2cceeea1d3be410a54356e5953008cb1f2f2f367 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -404,6 +404,18 @@ else ifeq ($(platform), miyoo)
    PLATFORM_DEFINES := -DDINGUX
    CFLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s
    CXXFLAGS += $(CFLAGS)
+   
+# TRIMUI MODEL S
+else ifeq ($(platform), trimui)
+  TARGET := $(TARGET_NAME)_libretro.so
+  CC  = $(CROSS_COMPILE)gcc
+  CXX = $(CROSS_COMPILE)g++
+  AR  = $(CROSS_COMPILE)ar
+  fpic := -fPIC
+  SHARED := -shared -Wl,-version-script=$(version_script)
+  PLATFORM_DEFINES := -DDINGUX
+  CFLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s
+  CXXFLAGS += $(CFLAGS)
 
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
