diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index caa22721a8efd5fcfb05d54845d69649a57149ed..d41a29a38aa50eaf7d81062e9b09f9e499e3787d 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -492,6 +492,19 @@ else ifeq ($(platform), miyoo)
 	HAVE_ARMv6 = 0
 	LOW_MEMORY = 1
 
+# TRIMUI MODEL S
+else ifeq ($(platform), trimui)
+  TARGET := $(TARGET_NAME)_libretro.so
+  CC  = $(CROSS_COMPILE)gcc
+  CXX = $(CROSS_COMPILE)g++
+  AR  = $(CROSS_COMPILE)ar
+  SHARED := -shared -nostdlib
+  fpic := -fPIC
+  LIBM :=
+  CFLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s -D__GCW0__
+  HAVE_ARMv6 = 0
+  LOW_MEMORY = 1
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 	NO_GCC := 1
