diff --git a/Makefile.libretro b/Makefile.libretro
index 40d635b..64fa91f
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -504,6 +504,25 @@ else ifeq ($(platform), rpi4)
 	BUILTIN_GPU = neon
 	DYNAREC = ari64
 
+# TRIMUI SMART PRO
+else ifeq ($(platform), tg5040)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	CFLAGS += -mtune=cortex-a53 -mcpu=cortex-a53 -march=armv8-a+simd -fomit-frame-pointer
+	CFLAGS += -ffast-math -fdata-sections -ffunction-sections -fsingle-precision-constant -flto -fPIC
+	LDFLAGS += -flto -fPIC
+	ifeq (,$(DEBUG))
+		# LDFLAGS += -s
+	endif
+	CPU_ARCH := arm64
+	OPTIMIZE := -O3 -Ofast -DNDEBUG=1
+	HAVE_NEON = 1
+	BUILTIN_GPU = neon
+	ARCH = arm64
+	DYNAREC = ari64
+
 #Raspberry Pi 4 with a 64bit GNU/Linux OS
 else ifeq ($(platform), rpi4_64)
         TARGET := $(TARGET_NAME)_libretro.so
