diff --git a/Makefile.libretro b/Makefile.libretro
index 2197946c..df72d03d
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -518,6 +518,25 @@ else ifeq ($(platform), rpi4_64)
         fpic := -fPIC
         CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a72 -ftree-vectorize
 
+# TRIMUI SMART PRO
+else ifeq ($(platform), tg5040)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	CFLAGS += -mtune=cortex-a53 -mcpu=cortex-a53 -march=armv8-a+simd -fomit-frame-pointer
+	CFLAGS += -ffast-math -fdata-sections -ffunction-sections -fsingle-precision-constant -flto -fPIC
+	LDFLAGS += -flto -fPIC
+	ifeq (,$(DEBUG))
+		# LDFLAGS += -s
+	endif
+	CPU_ARCH := arm64
+	OPTIMIZE := -O3 -Ofast -DNDEBUG
+	HAVE_NEON = 1
+	BUILTIN_GPU = neon
+	ARCH = arm64
+	DYNAREC = ari64
+	
 # Classic Platforms ####################
 # Platform affix = classic_<ISA>_<µARCH>
 # Help at https://modmyclassic.com/comp
