diff --git a/Makefile b/Makefile
index 3d068a5..7d14f14 100644
--- a/Makefile
+++ b/Makefile
@@ -378,7 +378,24 @@ else ifeq ($(platform), miyoo)
    SHARED := -shared -Wl,--version-script=libretro/link.T -Wl,-no-undefined
    CFLAGS += -ffast-math -march=armv5te -mtune=arm926ej-s -fomit-frame-pointer
    HAVE_LOW_MEMORY = 1
-
+# target platform tg5040 trimUI brick
+else ifeq ($(platform), tg5040)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   fpic := -fPIC
+   LDFLAGS += -s $(fpic) -shared -Wl,--version-script=libretro/link.T -Wl,-no-undefined
+   CFLAGS += -mtune=cortex-a53 -mcpu=cortex-a53 -march=armv8-a
+   CFLAGS += -fomit-frame-pointer -ffast-math -fPIC -flto
+   $(MYARCH)
+   CXXFLAGS += $(CFLAGS)
+   CPPFLAGS += $(CFLAGS)
+   ASFLAGS += $(CFLAGS)
+   HAVE_NEON = 1
+   HAVE_LOW_MEMORY = 1
+   CPU_ARCH := arm64
+   ARM = 1
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
 TARGET := $(TARGET_NAME)_libretro_xdk1.lib
