diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index 14b8cbedb12860aedc2bc21119735247929bdbd3..0d1787ef81ffc6cdb2b4982c779a89ef198c954f 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -461,6 +461,20 @@ else ifeq ($(platform), miyoo)
 	PLATFORM_DEFINES += -fomit-frame-pointer -ffast-math -mcpu=arm926ej-s
 	EXTERNAL_ZLIB = 1
 
+# RG35XX
+else ifeq ($(platform), rg35xx)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   SHARED := -shared -Wl,--version-script=src/libretro/link.T -Wl,-no-undefined
+   LDFLAGS += -fPIC -flto
+   CFLAGS += -marm -mtune=cortex-a9 -mfpu=neon-fp16 -mfloat-abi=hard -march=armv7-a -fomit-frame-pointer
+   CFLAGS += -fomit-frame-pointer -ffast-math -fPIC -flto
+   OPTIMIZE := -Ofast -DNDEBUG
+   EXTERNAL_ZLIB = 1
+   HAVE_NTSC = 0
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 	CC  = cl.exe
